bot/utils/checks.py

```diff
@@ -60,6 +61,10 @@ def in_whitelist_check(
         # categories, it's probably not wise to rely on its category in any case.
         channels = tuple(channels) + (redirect,)
 
+    # If it's a thread, and its parent is whitelisted, we whitelist the thread as well
+    if channels and isinstance(ctx.channel, Thread) and ctx.channel.parent_id in channels:
```

Comment:
Rather than adding the thread to the whitelisted channels, it makes more sense to check the parent channel instead of ctx.channel in this case. This also keeps the `in channels` check in one place.

Reply:
fwiw I usually do a hasattr check whenever I have to do something like this, and save the extra import, but that's more a matter of style.

Reply:
I just wanted to avoid multiple ifs within each other, but i can do that if you want